### PR TITLE
[XXX-55386] Change nets version to v10

### PR DESCRIPTION
**Pull request by monroid**
### Details:
 - inference-engine\tests\functional\inference_engine\local_test.cpp
 
### Tickets:
 - [XXX-55386]

### CODE REVIEW

**Comment by dmaksimo** [comment]
@monroid , let's use code review document with the relevant articles. Please add that to the description|:

https://wiki.ith.intel.com/display/CVSDK/Model+Optimizer+Code+Review+Template

**Comment by monroid** [comment]
@dmaksimo as I see, different teams use dedicated templates. My change relates to tests, so MO rules are really complicated for this one, IMHO.

File: inference-engine/tests/functional/inference_engine/local_test.cpp [hunk 1]
```diff
@@ -15,183 +15,636 @@ using namespace InferenceEngine;
...
-                </port>
-            </output>
-        </layer>
-    </layers>
-    <edges>
-        <edge from-layer="0" from-port="0" to-layer="1" to-port="1"/>
-        <edge from-layer="1" from-port="2" to-layer="2" to-port="1"/>
-        <edge from-layer="0" from-port="0" to-layer="2" to-port="2"/>
-    </edges>
-    <pre-process reference-layer-name="data">
```
Comment (vurusovs) [inline inference-engine/tests/functional/inference_engine/local_test.cpp:83]: @ilya-lavrenov is there a need to add pre-process section to check it's correctness?

File: inference-engine/tests/functional/inference_engine/local_test.cpp [hunk 2]
```diff
@@ -15,183 +15,636 @@ using namespace InferenceEngine;
...
-            <mean value="104.006"/>
-            <scale value="0.1"/>
-        </channel>
-        <channel id="1">
-            <mean value="116.668"/>
-            <scale value="0.2"/>
-        </channel>
-        <channel id="2">
-            <mean value="122.678"/>
-            <scale value="0.3"/>
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/local_test.cpp:94]: we need IR with floating point values while new V10 version does not have it

File: inference-engine/tests/functional/inference_engine/local_test.cpp [hunk 3]
```diff
@@ -15,183 +15,636 @@ using namespace InferenceEngine;
...
-        <edges>
-            <edge from-layer="0" from-port="0" to-layer="1" to-port="0"/>
-            <edge from-layer="1" from-port="1" to-layer="2" to-port="0"/>
-            <edge from-layer="1" from-port="2" to-layer="2" to-port="1"/>
-            <edge from-layer="1" from-port="3" to-layer="2" to-port="2"/>
-            <edge from-layer="2" from-port="3" to-layer="3" to-port="0"/>
-            <edge from-layer="2" from-port="4" to-layer="3" to-port="1"/>
-        </edges>
-    </net>
+<net name="model" version="10">
```
Comment (iefode) [inline inference-engine/tests/functional/inference_engine/local_test.cpp:101]: a little long IR, is it possible to decrease model? To remove extra ops for example?

File: inference-engine/tests/functional/inference_engine/local_test.cpp [hunk 4]
```diff
@@ -15,183 +15,636 @@ using namespace InferenceEngine;
...
+					<dim>12</dim>
+				</port>
+			</input>
+		</layer>
+	</layers>
+	<edges>
+		<edge from-layer="0" from-port="0" to-layer="1" to-port="0"/>
+		<edge from-layer="1" from-port="1" to-layer="2" to-port="0"/>
+	</edges>
+	<meta_data>
```
Comment (iefode) [inline inference-engine/tests/functional/inference_engine/local_test.cpp:56]: we can remove `<meta_data>` i think

**Comment by dmaksimo** [comment]
> @dmaksimo as I see, different teams use dedicated templates. My change relates to tests, so MO rules are really complicated for this one, IMHO.

But can you add a description, RCA and Solution test, so the reader can understand the changes?

File: inference-engine/tests/functional/inference_engine/local_test.cpp (hunk 2 above)
Comment (monroid) [reply inference-engine/tests/functional/inference_engine/local_test.cpp:94]: added

File: inference-engine/tests/functional/inference_engine/local_test.cpp (hunk 3 above)
Comment (monroid) [reply inference-engine/tests/functional/inference_engine/local_test.cpp:95]: This one is the shortest, we've alo tried TW LSTM it was a huge one.

File: inference-engine/tests/functional/inference_engine/local_test.cpp (hunk 4 above)
Comment (monroid) [reply inference-engine/tests/functional/inference_engine/local_test.cpp]: done

**Review by iefode — APPROVED** [review]

File: inference-engine/tests/functional/inference_engine/local_test.cpp [hunk 5]
```diff
@@ -15,183 +15,589 @@ using namespace InferenceEngine;
...
-            <edge from-layer="0" from-port="0" to-layer="1" to-port="0"/>
-            <edge from-layer="1" from-port="1" to-layer="2" to-port="0"/>
-            <edge from-layer="1" from-port="2" to-layer="2" to-port="1"/>
-            <edge from-layer="1" from-port="3" to-layer="2" to-port="2"/>
-            <edge from-layer="2" from-port="3" to-layer="3" to-port="0"/>
-            <edge from-layer="2" from-port="4" to-layer="3" to-port="1"/>
-        </edges>
-    </net>
+<net name="model" version="10">
+	<layers>
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/local_test.cpp:96]: I don't see floating point numbers in LSTM model

Comment (monroid) [reply inference-engine/tests/functional/inference_engine/local_test.cpp:96]: added

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/tests/functional/inference_engine/local_test.cpp [hunk 6]
```diff
@@ -205,40 +226,23 @@ class LocaleTests : public ::testing::Test {
...
-            InferenceEngine::NetPass::UnrollRNN_if(net, [] (const RNNCellBase& rnn) -> bool { return true; });
-            auto lstmcell_layer = dynamic_pointer_cast<ClampLayer>(CommonTestUtils::getLayerByName(net, "LSTMCell:split_clip"));
-
-            float ref_coeff = 0.2f;
-            ASSERT_EQ(lstmcell_layer->min_value, -ref_coeff);
-            ASSERT_EQ(lstmcell_layer->max_value,  ref_coeff);
-
-            ASSERT_EQ(lstmcell_layer->GetParamAsFloat("min"), -ref_coeff);
-            ASSERT_EQ(lstmcell_layer->GetParamAsFloat("max"),  ref_coeff);
+            auto lstmcell_layer = CommonTestUtils::getLayerByName(InferenceEngine::CNNNetwork(convertedNetwork), "LSTMCell");
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/local_test.cpp:243]: `getLayerByName` is deprecated API. We must use ngraph w/o conversion to `CNNNetworkImpl`

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/tests/functional/inference_engine/local_test.cpp [hunk 7]
```diff
@@ -205,40 +224,29 @@ class LocaleTests : public ::testing::Test {
...
+                    const auto roi = std::dynamic_pointer_cast<ngraph::op::v3::ROIAlign>(op);
+                    ASSERT_EQ(roi->get_pooled_h(), 7);
+                    ASSERT_EQ(roi->get_pooled_w(), 7);
+                    ASSERT_EQ(roi->get_sampling_ratio(), 2);
+                    ASSERT_EQ(roi->get_spatial_scale(), 0.25f);
+                }
+            } else {
+                if (op->get_friendly_name() == "LSTMCell") {
+                    const auto lstm_seq = std::dynamic_pointer_cast<ngraph::op::util::RNNCellBase>(op);
+                    ASSERT_EQ(lstm_seq->get_clip(), 0.0f);
```
Comment (vurusovs) [inline inference-engine/tests/functional/inference_engine/local_test.cpp:246]: let's check `hidden_size` attribute too

**Review by vurusovs — APPROVED** [review]
LGTM!

**Comment by monroid** [comment]
@ilya-lavrenov @iefode @vurusovs Thanks, folks!

**Pull request merged**